fix(adapter): transistion to surrealdb@^1.0.0

dvanmali opened the pull request:
## ☕️ Reasoning

Moves package dependency to version 1 of the surrealdb javascript library. Package includes websocket connectivity, type safety, updates to the setup documentation, and package peer dependency change.

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [x] Ready to be merged

## 🎫 Affected issues

#11732 - solves this by transistioning to surrealdb@v1.0.0 package. Docs state to use new connection string.
#11866 - javascript package supports surrealdb v2
#11471 - uses current surrealdb software

## 📌 Resources

- [Surrealdb.js](https://surrealdb.com/docs/sdk/javascript/core)

dvanmali committed:
feat: transition from surrealdb.js beta to surrealdb@1.0.0

dvanmali committed:
fix: ensure correct adapter type is returned with verified is nulled if not provided

dvanmali committed:
merge: fix conflicts from conflicts

dvanmali committed:
fix: remove log print on test

dvanmali committed:
feat: more AnyAuth login variables, support HTTP rpc connection, add exponential backoff for disconnection events to example

dvanmali committed:
merge: with upstream/main

dvanmali committed:
Merge branch 'main' of github.com:nextauthjs/next-auth

dvanmali committed:
restore: changes

dvanmali committed:
restore: changes

dvanmali committed:
doc: add all env variables available

dvanmali committed:
fix: throw error not string

liamwh commented:
Any updates on this PR? Would love to see it merged so we can use SurrealDB with Auth.js!

dvanmali committed:
merge: upstream main

applied-mathematician commented:
Bump.  I'm waiting for this PR to use Auth.js since I'm using Surrealdb.

dvanmali committed:
Merge branch 'main' into main

dvanmali committed:
Merge branch 'main' into main

ndom91 commented:
This is breaking, right? Well have to make sure we bump the surreal adapter appropriately

dvanmali commented:
> [ndom91](https://github.com/ndom91) This is breaking, right? Well have to make sure we bump the surreal adapter appropriately

Yes, this is a breaking update because the underlying storage layer and transmission layer are now both typed. Thus this package jumps to version 2.0.0 with surrealdb@^1.0.0

ctison commented:
Surreal Cloud just launched! I am planning to use it everywhere I can. Do you know what is blocking this PR?

dvanmali committed:
Merge branch 'main' into main

Padi2312 commented:
Is this PR still up-to-date ? Looking forward to use SurrealDB and AuthJS together  but with surrealdb v2 it doesnt work anymore.

dvanmali commented:
> Is this PR still up-to-date ? Looking forward to use SurrealDB and AuthJS together but with surrealdb v2 it doesnt work anymore.

Yes this PR should still work with v2, if you have issues let me know

Padi2312 commented:
@dvanmali sorry for my bad english, i ment whether this PR will ever get merged :D

I just copied your index.ts file and imported from local package, works perfectly fine. Thanks🫱🏼‍🫲🏽

dvanmali committed:
Merge branch 'main' into main

dvanmali commented:
@ndom91 @0ubbe PR updates? It's been 5 months since opening

dvanmali committed:
Merge branch 'main' into main

GeoDoX commented:
@ThangHuuVu Since this is a blocking PR, can we get this reviewed and merged ASAP please?

ThangHuuVu committed:
Merge branch 'main' into main

ThangHuuVu committed:
Merge branch 'main' into main

ThangHuuVu reviewed (approved):
Thanks for putting this together! the tests are passing on my end, but during testing I need to use the v1.5 tag for the docker image, as the latest tag gives me this error:

```
UnsupportedVersion: The version "1.0.0+20230913.54aedcd" reported by the engine is not supported by this library, expected a version that satisfies ">= 1.4.2 < 3.0.0".
```
Lets see if the CI test passes

ThangHuuVu commented:
@dvanmali the lockfile needs to be updated, could u help with that? You are creating the PR from your `main` which makes it a bit hard for maintainers to push changes to.

dvanmali committed:
merge: upstream

dvanmali committed:
fix: uses surrealdb:^v1.3.0 for native reconnectivity, feat: add webauthn funcs to adapter

dvanmali committed:
fix: creating user should use id() not randomUUID()

dvanmali committed:
fix: ignored files from prettier

dvanmali committed:
chore: upgrade lockfile as requested

dvanmali commented:
Sure @ThangHuuVu, just updated the lockfile and fixed the shell script so it can now pass your test scripts. It appears that Surrealdb v3-alpha is currently tied to latest so it may not currently be compatible as they mention on their [javascript library](https://surrealdb.com/docs/sdk/javascript#javascript-sdk).

dvanmali commented:
Feel free to message this thread again if you wish for another update or add a PR to my main. Unfortunately because I don't have write access to this repo, i can't make a branch here, hence the forked repo from main

dvanmali committed:
Merge branch 'main' into main

dvanmali commented:
Were you able to take a look @ThangHuuVu?

ThangHuuVu commented:
@dvanmali just approved the jobs but the build is failing, could u help check it? 🙏

dvanmali committed:
Merge branch 'main' of github.com:nextauthjs/next-auth

dvanmali committed:
fix: tsc build errors

dvanmali committed:
Merge branch 'main' of github.com:dvanmali/next-auth

dvanmali committed:
fix: tsc build errors

dvanmali commented:
Good to go @ThangHuuVu

dvanmali commented:
Gtg @ThangHuuVu 👍

dvanmali committed:
Merge branch 'main' into main

dvanmali commented:
@ndom91 @balazsorban44 @ThangHuuVu gtg

ndom91 committed:
Merge branch 'main' into main

dvanmali commented:
@ndom91 @balazsorban44 @ThangHuuVu Vercel keeps blocking the build with "authorization required to deploy"

dvanmali commented:
Love it! Thank you @ThangHuuVu

GeoDoX commented on packages/adapter-surrealdb/src/index.ts:
```diff
@@ -14,138 +14,197 @@
+          return docToUser(userDoc[0])
+        }
+      } catch {}
       throw new Error("User not created")
```
@dvanmali Finally getting around to try out your changes and noticed something and had a suggestion. this error always gets thrown. If you have time it would also be great if you could use the logger to log the errors being caught as well.

geodox replied:
Will do, I should have some time tonight or tomorrow. Thanks!

geodox replied:
I missed the return just above, so it turns out the error doesn't always get thrown. I looked into the logger, and I guess it's not easily accessible. Some better logging would be helpful though, but perhaps that's more on AuthJS side as a stacktrace doesn't even get logged.

geodox commented:
@dvanmali Can you provide SurrealQL code to setup schemafull tables?
